core/trino-main/src/main/java/io/trino/testing/StandaloneQueryRunner.java

```diff
@@ -65,6 +67,8 @@ public final class StandaloneQueryRunner
     private final InMemorySpanExporter spanExporter = InMemorySpanExporter.create();
 
     private final ReadWriteLock lock = new ReentrantReadWriteLock();
+    private final AtomicInteger concurrentQueries = new AtomicInteger();
+    private boolean spansValid = true;
```

Comment:
What guards accesses to this non-volatile field?

Reply:
good point.
i can make it volatile, but it doesn't make this bullet proof anyway.
so conceptually this field is used normally in single-threaded scenario only (the only scenario where it's legit to run query and then ask for spans).

Reply:
https://github.com/trinodb/trino/pull/20796